src/sparse_int_set.jl

```diff
@@ -0,0 +1,258 @@
+import Base: @propagate_inbounds, zip
+
+const INT_PER_PAGE = div(ccall(:jl_getpagesize, Clong, ()), sizeof(Int))
+
+#TODO: Batch creation and allocation
+mutable struct SparseIntSet
+	packed ::Vector{Int}
+	reverse::Vector{Vector{Int}}
+	function SparseIntSet()
+		out = new(Int[], Vector{Int}[])
+		return out
+	end
+end
+
+function SparseIntSet(indices)
+	set = SparseIntSet()
+	for i in indices
+		i <= 0 && throw(ArgumentError("All Ints need to be positive."))
+		push!(set, i)
+	end
+	return set
+end
+
+SparseIntSet(i...) = SparseIntSet(i)
+
+eltype(::SparseIntSet) = Int
+eltype(::Type{SparseIntSet}) = Int
+empty(::SparseIntSet) = SparseIntSet()
+isempty(s::SparseIntSet) = isempty(s.packed)
+
+copy(s::SparseIntSet) = copy!(SparseIntSet(), s)
+
+function copy!(to::SparseIntSet, from::SparseIntSet)
+	to.packed = copy(from.packed)
+	to.reverse = deepcopy(from.reverse)
```

Comment:
This will need to change once we switch to outer constructors + immutable structs